app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -64,27 +77,80 @@ internal class DefaultMetricsStorage(
             Event.GrowthData.FirstUriLoadForDay -> {
                 settings.uriLoadGrowthLastSent = System.currentTimeMillis()
             }
+            Event.GrowthData.FirstWeekSeriesActivity -> {
+                settings.firstWeekSeriesGrowthSent = true
+            }
         }
     }
 
+    private fun updateDaysOfUse() {
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData
+        val currentDate = Calendar.getInstance()
+        val asString = dateFormatter.format(currentDate.time)
+        if (currentDate.timeInMillis.withinFirstWeek() && daysOfUse.none { it == asString }) {
+            settings.firstWeekDaysOfUseGrowthData = daysOfUse.plus(asString)
+        }
+    }
+
+    private fun shouldTrackFirstWeekActivity(): Boolean = Result.runCatching {
+        if (!System.currentTimeMillis().withinFirstWeek() || settings.firstWeekSeriesGrowthSent) {
+            return false
+        }
+
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData.map {
+            dateFormatter.parse(it)
+        }.sorted()
+
+        // This loop will check whether the existing list of days of use, combined with the
+        // current date, contains any periods of 3 days of use in a row. It's mostly this ugly
+        // because of the inherent mutability of the Calendar type, but hopefully we can rip
+        // this code out soon™ anyway
+        for (idx in daysOfUse.indices) {
+            if (idx + 1 > daysOfUse.lastIndex || idx + 2 > daysOfUse.lastIndex) {
+                continue
```

Comment:
thoughts on using the `continue` vs wrapping the entire block inside the `if` instead?

Reply:
I can change it if you'd prefer. I just try to avoid indentation where possible in case of awkward line breaks

Reply:
The hesitation to use `continue` is mostly a personal coding preference of mine. It would have been more intuitive if it had been possible to do an iOS equivalent of `guard` with a variable and just return/continue if a variable couldn't be initialized, but I tried and couldn't get it to work. 

I think both are readable; I just think `continue` can make _some_ blocks of code less intuitive. I think it's fine here.